src/auto-trace.js

```diff
@@ -74,34 +85,72 @@ export function syncStacktrace(rawError, ...extraMiddlewares){
  * @param {function (Object err) => Object newErr } [extraMiddlewares] - middleware function(s) that will be called on rawError before being handled
  * @throws {Error} 
  */
-export function throwSyncStacktrace(rawError, ...extraMiddlewares){
-	throw syncStacktrace(rawError, ...extraMiddlewares);
+export function throwSyncStacktrace() {
+	const asyncStacktraceErr = new Error();
+	const syncMiddlewareErrFunctions = executeAsyncMiddleware(asyncStacktraceErr);
+	
+	return (rawError) => {
+		const syncErr = rawError || new Error()
+		const middlewareErr = executeSyncMiddleware(syncMiddlewareErrFunctions, syncErr);
+		throw wrapObjectWithError(middlewareErr, syncErr);
+	};
 }
 
 /**
  * Calls globalMiddleware and extraMiddleware functions on rawError
- * Does not create a stacktrace nor will it wrap rawError in Error object
+ * Does not create a stacktrace
  *
  * @param {Object}   [rawError] 
  * @param {function (Object err) => Object newErr } [extraMiddlewares] - middleware function(s) that will be called on rawError
  * @returns {Object} 
  */
-export function withoutStacktrace(rawError, ...extraMiddlewares){
-	const middlewareErr = executeMiddleware(rawError, extraMiddlewares);
-	return middlewareErr;
+export function withoutStacktrace(callback = ()=>{}) {
```

Comment:
also a higher order function - same reason as syncStacktrace. Makes this function more useful as this one allows the middleware to determine what the stacktrace will be.